templates/master.erb

```diff
@@ -1,29 +1,13 @@
-# ZooKeeper URL (used for leader election amongst masters)
-ZK="<%= @zookeeper %>"
+MESOS_work_dir="<%= @work_dir %>"
+MESOS_zk="<%= @zookeeper %>"
+MESOS_quorum=<%= @quorum %>
+MESOS_ip="<%= @listen_address %>"
+MESOS_port=<%= @master_port %>
+MESOS_cluster="<%= @cluster %>"
+MESOS_log_dir="<%= @log_dir %>"
+MESOS_logbufsecs=1
+MESOS_logging_level="INFO"
```

Comment:
@deric I hope I understand your comment correctly.
As this is a template it could be used to generate defaults or other files.
I'm suggesting it should be based on the environment variables and, hence could be sourced (e.g. by systemd). 

The current one relies on variables that are replaced with environment variables by a script that is not versioned with mesos (not even source if it's available).

IMHO there are too many ways to configure things (this, defaults, configs, files, env vars and cli args) and this could be confusing and also unneeded in Puppet managed environments.